src/values.test.ts

```diff
@@ -53,6 +53,14 @@ describe("typing", () => {
     expectTypeOf(result).toEqualTypeOf<Array<boolean>>();
   });
 
+  it("should correctly type union of Records", () => {
+    const result = values({
+      a: "b",
+    } as Record<number, string> | Record<string, string>);
+
+    expectTypeOf(result).toEqualTypeOf<Array<string>>();
```

Comment:
This doesn't test the main issue this PR deals with clearly enough; the records should have different value types so that we can see that the result is the correct union of them; using string for both just skips the complexity, and it doesn't matter what the key is.
```suggestion
    const result = values({
      a: "b",
    } as Record<PropertyKey, "cat"> | Record<PropertyKey, "dog">);

    expectTypeOf(result).toEqualTypeOf<Array<string>>();
```